### DISCUSSION TITLE
Release: pre-compiled binaries

### DISCUSSION THREAD

**Discussion by ThomasGreenhill** [python-control/python-control#741] [Q&A] [answered]
Is there a reason why you guys don't release pre-compiled binaries?

**Comment by bnavigator** [comment, marked as answer]
Hi,

1. We publish conda packages on conda-forge. Anaconda is the recommended ecosystem for scientific Python users on Windows and Mac.
2. Because python-control is a "pure" python library, there is nothing to compile. There is no difference between installing the source or a wheel from PyPI (`pip install control`)
3. For a discussion, why we didn't publish any Slycot wheels yet, see https://github.com/python-control/Slycot/issues/155

**Reply by ThomasGreenhill** [reply]
Thank you for the response! 
I ask because we use a hermetic bazel environment, and running a `setup.py` breaks that hermicity. 
I'll find a workaround on my end, thanks for the response

**Reply by bnavigator** [reply]
Interesting. In my efforts to package various rpm packages I had to fight Bazel for the opposite case: It wan'ts to download stuff all the time.

I guess you can tell Bazel to get everything you need beforehand:

- Install OpenBLAS (or your preferred LAPACK/BLAS implementation), CMake, scikit-build
- build and install Slycot
- install python-control.

**Reply by ilayn** [reply]
In its current state, there is no straightforward way to install Slycot on windows at all. And probably the reason why Slycot wheels don't exist on Windows. 

> scientific Python users on Windows and Mac.

I'd precise that by saying, this is a recommendation from the python-control maintainer perspective.

**Reply by bnavigator** [reply]
The same recommendation is found at https://scipy.org/install/ btw.

**Reply by ilayn** [reply]
Yes true. Those pages are going away very soon also reflects the maintainer view from a decade ago (for example winpython is not a thing anymore etc.) but we went over and beyond to make the binaries available for most systems including windows users. The problem is the lack of a good Fortran toolchain which we hacked through MSVC + gfortran from Mingw project. And that is the actual reason not necessarily python-control's or SLICOT devs' fault.
